zebrad/src/config.rs

```diff
@@ -59,6 +59,28 @@ pub struct TracingSection {
 
     /// The endpoint address used for tracing.
     pub endpoint_addr: SocketAddr,
+
+    /// The path to write a flamegraph of tracing spans too.
+    ///
+    /// This path is not used verbatim when writing out the flamegraph. This is
+    /// because the flamegraph is written out as two parts. First the flamegraph
+    /// is constantly persisted to the disk in a "folded" representation that
+    /// records collapsed stack traces of the tracing spans that are active.
+    /// Then, when the application is finished running the destructor will flush
+    /// the flamegraph output to the folded file and then read that file and
+    /// generate the final flamegraph from it as an SVG.
+    ///
+    /// The need to create two files means that we will slightly manipulate the
+    /// path given to us to create the two representations.
+    ///
+    /// # Example
+    ///
+    /// Given `flamegraph = "flamegraph"` we will generate a `flamegraph.svg`
+    /// and a `flamegraph.folded` file in the current directory.
+    ///
+    /// If you provide a path with an extension the extension will be ignored and
+    /// replaced with `.folded` and `.svg` for the respective files.
```

Comment:
Thanks for documenting this behaviour.

Most unix tools would just append the extra extension to any existing extension, so some developers might be surprised that we strip the existing extension.

(To be clear, I don't think this behaviour needs to change.)